drivers/gpu/drm/vc4/vc4_hdmi.c

```diff
@@ -448,6 +458,17 @@ static void vc4_hdmi_handle_hotplug(struct vc4_hdmi *vc4_hdmi,
 	if (!edid)
 		return;
 
+	/*
```

Comment:
Do we need to separate the plugged and unplugged events either side of the EDID parsing, or can we just always report them after (ie from here)?

Reply:
I'd missed the return.

Possibly just split the edid fragment?
```
	edid = drm_get_edid(connector, vc4_hdmi->ddc);

	/* We need to process the EDID before calling the plugged callback */
	if (status == connector_status_connected) {
		mutex_lock(&vc4_hdmi->update_plugged_status_lock);
		if (vc4_hdmi->plugged_cb && vc4_hdmi->codec_dev)
			vc4_hdmi->plugged_cb(vc4_hdmi->codec_dev, CONDITIONAL);
		mutex_unlock(&vc4_hdmi->update_plugged_status_lock);
	}

	if (!edid)
		return;
```

Reply:
I've done essentially this but also put an `if(connected)` before fetching the EDID.  I've convinced myself that we can't uninitialised-read `edid` (though it's less obvious than I would like)

Reply:
You are correct that reading the EDID when disconnected would be an odd thing to do, so I'm happy with that.

Reply:
Is that going to be enough to satisfy all compilers? Wouldn't initialising edid to NULL be better?

Reply:
I've changed to initialise to NULL just in case.